frontend/gateway/pb/gateway.proto

```diff
@@ -63,6 +64,36 @@ message RefMap {
 	map<string, Ref> refs = 1;
 }
 
+message Attestations {
+	message Attestation {
+		oneof Attestation {
+			InToto intoto = 1;
+		}
+	}
+	repeated Attestation attestation = 1;
+}
+
+message InToto {
+	string predicateType = 1;
+	Ref predicateRef = 2;
+	string predicatePath = 3;
+
+	message Subject {
+		message SelfSubject {
+		}
+		message RawSubject {
```

Comment:
I'm not sure if allowing `RawSubject` isn't a security concern actually. If we let the frontend to claim that attestation is for a random digest there is no way BuildKit can validate this is any way. 

It is possible we should only allow attesting data when we can prove how it was built. Or at least we should make it so that it is clear that this subject is not allowed to point to any container objects so validators know that this is not to be trusted.

Something to pick up again on follow-ups.

Reply:
I think it depends on the security model that users are running buildkit under - but agreed.

We can revisit this, I think we could work round the need for a `RawSubject` by allowing attesting to individual files (composed of a `Ref` + `Path`), which is the main other use I can see.